File: java/src/main/java/com/autogenai/llmservice/service/stubs/BedrockClient.java
package com.autogenai.llmservice.service.stubs;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

@Service
public class BedrockClient {
    public record Delta(String text) {}
    public record ContentBlockDelta(Integer contentBlockIndex, Delta delta) {}

    public record ContentPart(String text) {}
    public record Message(String role, List<ContentPart> content) {}
    
    public record InferenceConfig(Optional<Double> temperature, Optional<Integer> maxTokens) {}
    
    public record BedrockRequest(
            String modelId,
            Optional<String> system,
            List<Message> messages,
            InferenceConfig inferenceConfig
    ) {}

    public record BedrockChunk(ContentBlockDelta contentBlockDelta) {}

    public Stream<BedrockChunk> streamChat(BedrockRequest request) {
        return Stream.of(
                "Hello ", "from ", "Bedrock"
        ).map(text -> new BedrockChunk(
                new ContentBlockDelta(0, new Delta(text))
        ));
    }
}

File: c#/llmservice/Service/stubs/BedrockClient.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace llmservice.Services.Stubs
{
    public interface IBedrockClient
    {
        IAsyncEnumerable<BedrockChunk> StreamChatAsync(BedrockRequest request);
        public record Delta(string Text);

        public record ContentBlockDelta(int ContentBlockIndex, Delta Delta);

        public record ContentPart(string Text);

        public record Message(string Role, List<ContentPart> Content);

        public record InferenceConfig(double? Temperature, int? MaxTokens);

        public record BedrockRequest(
            string ModelId,
            string? System,
            List<Message> Messages,
            InferenceConfig InferenceConfig
        );
        public record BedrockChunk(ContentBlockDelta ContentBlockDelta);
    }


    public class BedrockClient : IBedrockClient
    {
        public record Delta(string Text) { }

        public record ContentBlockDelta(int ContentBlockIndex, Delta Delta) { }

        public record ContentPart(string Text) { }

        public record Message(string Role, List<ContentPart> Content) { }

        public record InferenceConfig(double? Temperature, int? MaxTokens) { }

        public record BedrockRequest(
                string ModelId,
                string? System,
                List<Message> Messages,
                InferenceConfig InferenceConfig
            )
        { }
        public record BedrockChunk(ContentBlockDelta ContentBlockDelta) { }


        public async IAsyncEnumerable<IBedrockClient.BedrockChunk> StreamChatAsync(IBedrockClient.BedrockRequest request)
        {
            await Task.Delay(10);

            var textChunks = new List<string> { "Hello ", "from ", "Bedrock" };

            int index = 0;

            foreach (var text in textChunks)
            {
                await Task.Delay(50);

                yield return new IBedrockClient.BedrockChunk(
                    new IBedrockClient.ContentBlockDelta(index++, new IBedrockClient.Delta(text))
                );
            }
        }
    }
}